### PR TITLE
[SLA] PIM-7116: Enforce anti-CSRF security on internal API

### PR DESCRIPTION
**Pull request by tamarasaurus**
**Description (for Contributor and Core Developer)**

Updated the following controllers:

```
M	src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php
M	src/Pim/Bundle/EnrichBundle/Controller/GroupController.php
M	src/Pim/Bundle/EnrichBundle/Controller/GroupTypeController.php
M	src/Pim/Bundle/EnrichBundle/Controller/ProductController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/ApiClientController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/AssociationTypeController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/AttributeGroupController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/FamilyController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/FamilyVariantController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/GroupController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/GroupTypeController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/JobInstanceController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductModelController.php
M	src/Pim/Bundle/EnrichBundle/Controller/Rest/ValuesController.php
M	src/Pim/Bundle/ImportExportBundle/Controller/JobProfileController.php
M	src/Pim/Bundle/NotificationBundle/Controller/NotificationController.php
```

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | -
| Added legacy Behats               | -
| Added acceptance tests            | -
| Added integration tests           | -
| Changelog updated                 | -
| Review and 2 GTM                  | OK
| Micro Demo to the PO (Story only) | -
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Controller/ProductController.php [hunk 1]
```diff
@@ -11,6 +11,7 @@
 use Pim\Component\Catalog\Repository\ProductRepositoryInterface;
 use Pim\Component\Catalog\ValuesFiller\EntityWithFamilyValuesFillerInterface;
 use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\RedirectResponse;
```
Comment (juliensnz) [inline src/Pim/Bundle/EnrichBundle/Controller/ProductController.php]: is it used?

Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/Controller/ProductController.php]: nope, will remove

File: src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php [hunk 2]
```diff
@@ -267,6 +267,10 @@ public function indexAction()
      */
     public function createAction(Request $request, $parent = null)
     {
+        if (!$request->isXmlHttpRequest()) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php]: Are you sure you can still edit the categories with it ? If I remember well this part is still "old school" so the request will never be an XHR.

File: src/Pim/Bundle/EnrichBundle/Controller/GroupController.php [hunk 3]
```diff
@@ -87,6 +88,10 @@ public function __construct(
      */
     public function createAction(Request $request)
     {
+        if (!$request->isXmlHttpRequest()) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupController.php]: idem

File: src/Pim/Bundle/EnrichBundle/Controller/GroupTypeController.php [hunk 4]
```diff
@@ -79,6 +80,10 @@ public function __construct(
      */
     public function createAction(Request $request)
     {
+        if (!$request->isXmlHttpRequest()) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/GroupTypeController.php]: idem

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductModelController.php [hunk 5]
```diff
@@ -227,10 +233,14 @@ public function createAction(Request $request): JsonResponse
      *
      * @AclAncestor("pim_enrich_product_model_edit_attributes")
      *
-     * @return JsonResponse
+     * @return Response
      */
-    public function postAction(Request $request, int $id): JsonResponse
+    public function postAction(Request $request, int $id): Re
```
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductModelController.php]: typo

File: src/Pim/Bundle/ImportExportBundle/Controller/JobProfileController.php [hunk 6]
```diff
@@ -97,6 +98,10 @@ public function __construct(
      */
     public function createAction(Request $request)
     {
+        if (!$request->isXmlHttpRequest()) {
```
Comment (phaseinducer) [inline src/Pim/Bundle/ImportExportBundle/Controller/JobProfileController.php]: idem, not an XHR

File: src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php (hunk 2 above)
Comment (tamarasaurus) [reply src/Pim/Bundle/EnrichBundle/Controller/CategoryTreeController.php]: yeah you're right, will remove

**Pull request merged**